lib/hawkular/metrics/metric_api.rb

```diff
@@ -129,10 +129,13 @@ def create(definition)
 
       # Query metric definitions by tags
       # @param tags [Hash]
+      # @param options [Hash] Additional options to configure
+      # @option options [Boolean] :timestamps If timestamps should be included on the response. Defaults to true
       # @return [Array[MetricDefinition]]
-      def query(tags = nil)
+      def query(tags = nil, options = {})
+        timestamps = options[:timestamps] || true
```

Comment:
With this you are allowing the callers to inject random data to the query string of the URL.
What I mean, if I call:
`query(nil, { timestamps: "random data" })`

The `timestamps` variable will have "`random data`" value; thus, URL built will be `metrics/?timestamps=random data&type=metric_type`

What is even more weird is that these calls are equivalent:
`query(nil, { timestamps: true })`
`query(nil, { timestamps: false })`

Because:
`timestamps = false || true` will evaluate to true (because it's an OR!!)
`timestamps = true || true` will evaluate to true (because there is no choice, ha)

and final URL will be `metrics/?timestamps=true&type=metric_type` for both cases and will return timestamps.
So... I think this line is not correct at all. May be you explicitly want to test against `true`:
`timestamps = options[:timestamps] == true`